spec/support/runner.rb

```diff
@@ -285,7 +298,12 @@ def after_setup
       package_path = "#{File.expand_path("../../../../../../", __dir__)}/"
       report_path_digest = Digest::SHA256.hexdigest(package_path)
 
-      File.write("/tmp/appsignal-#{report_path_digest}-install.report", install_report)
+      install_report_path = "/tmp/appsignal-#{report_path_digest}-install.report"
+      if install_report?
+        File.write(install_report_path, install_report)
+      elsif File.exist?(install_report_path)
+        File.delete(install_report_path)
+      end
```

Comment:
Perhaps this block here, which is repeated in the Ruby runner, could be part of the default `after_setup` for a runner? After all, so far `after_setup` is only being used for this. Something like:

```ruby
class Runner
  def after_setup
    if respond_to?(:install_report) and respond_to?(:install_report_path)
      # put the if-elsif-end block from above here
    end
  end
  
  class Nodejs < Runner # or Ruby < Runner
    def install_report_path
      # the logic to figure out the report path from the package path goes here
    end
    
    def install_report
      # this already exists
    end
  end
end
```

Reply:
Yes we should do something like this, but it was Friday when I wrote it and wanted to fix it quickly 😅

I'll have a look if this can be done better.

Reply:
That's fair 👍 It's a nice-to-have, definitely not a must-have.